youssef-mostafa25/pokedex
# Refactor class pokemon

### Before submitting this PR make sure:

- [x] build command ran successfully.
- [x] Your code does exactly what it is supposed to do.

### Description:

#### 1. Overview description:

I refactored my code.

#### 2. Changes made:

- Extracting all the pokeapi functionality into an interface.
- Adding a class that implements the interface.
- Extracting a class Pokemon from my code.

### Blockers:

None

### Manual test:

- [x] Make sure you've manually tested your code.
I tested all the use case scenarios to make sure everything was working as expected.

### How to test:

Test the use case scenarios

### Self review your code:


### Update the documentations:

- [x] Make sure you've updated the Readme.md, todo.txt and use_case.txt files.

### Write automatic tests:

- [ ] Make sure you've written automatic tests if needed.
- Write the locations of the test files, and mention their names?
